common/src/main/java/com/revenuecat/purchases/common/logWrapper.kt

```diff
@@ -0,0 +1,51 @@
+package com.revenuecat.purchases.common
+
+import com.revenuecat.purchases.strings.Emojis
+
+fun log(type: LogIntent, level: LogLevel, message: String) {
+    val emojifiedMessage = "${type.emojiList.joinToString("")} $message"
+    when (level) {
+        LogLevel.DEBUG -> debugLog(emojifiedMessage)
+        LogLevel.ERROR -> errorLog(emojifiedMessage)
+        LogLevel.INFO -> infoLog(emojifiedMessage)
+        LogLevel.WARN -> warnLog(emojifiedMessage)
+    }
+}
+
+/**
+ * Enum of emojis for log messages according to intent.
+ */
+enum class LogIntent(val emojiList: List<String>) {
+    /**
+     * Emojis for Google error messages.
+     */
+    GOOGLEERROR(listOf(Emojis.GOOGLE_ERROR, Emojis.DOUBLE_EXCLAMATION)),
+    /**
+     * Emoji for information messages.
+     */
+    INFO(listOf(Emojis.INFO)),
```

Comment:
@vegaro mentioned in a previous comment that single elements shouldn't be placed in a list. Since then, I added an enum instead. This made more sense to me to keep the implementation simple with `log` in `logWrapper`.

Reply:
@aboedo This definitely felt a bit awkward to me as well. Currently in the SDK revamp logging spreadsheet, we have some Google errors being logged as both `error` and `info` level. 

Example: `error`: "Error fetching products", but then we also have an `info` level: "Couldn't find existing purchase for sku". 

The reason for this was because Ryan & I wanted to indicate the source of the message. Unless we create an additional enum value called `GoogleInfo` and enforce the correlation between `LogIntent` and `LogLevel`.